test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -315,277 +315,445 @@ test.describe( 'Navigation block', () => {
+	test.describe( 'Navigation block focus management', () => {
+		test.beforeEach(
+			async ( { admin, editor, requestUtils, navigation } ) => {
+				await admin.createNewPost();
 
-		await requestUtils.createNavigationMenu( {
-			title: 'Animals',
-			content: '',
-		} );
+				await requestUtils.createNavigationMenu( {
+					title: 'Animals',
+					content: '',
+				} );
 
-		await editor.insertBlock( { name: 'core/navigation' } );
+				await editor.insertBlock( { name: 'core/navigation' } );
 
-		const navBlock = navigation.getNavBlock();
+				const navBlockInserter = navigation.getNavBlockInserter();
+				// Wait until the nav block inserter is visible before we continue. Otherwise the navigation block may not have finished being created.
+				await expect( navBlockInserter ).toBeVisible();
+			}
+		);
 
-		const navBlockInserter = navigation.getNavBlockInserter();
-		// Wait until the nav block inserter is visible before we continue. Otherwise the navigation block may not have finished being created.
-		await expect( navBlockInserter ).toBeVisible();
+		test( 'Focus management when using the navigation link appender', async ( {
+			pageUtils,
+			navigation,
+		} ) => {
+			const navBlock = navigation.getNavBlock();
 
-		/**
-		 * Test: We don't lose focus when using the navigation link appender
-		 */
-		await pageUtils.pressKeys( 'ArrowDown' );
-		await navigation.useBlockInserter();
-		await navigation.addLinkClose();
-		/**
-		 * TODO: This is not desired behavior. Ideally the
-		 * Appender should be focused again since it opened
-		 * the link control.
-		 * IMPORTANT: This check is not to enforce this behavior,
-		 * but to make sure focus is kept nearby until we are able
-		 * to send focus to the appender.
-		 */
-		await expect( navBlock ).toBeFocused();
+			/**
+			 * Test: We don't lose focus when using the navigation link appender
+			 */
+			await pageUtils.pressKeys( 'ArrowDown' );
+			await navigation.useBlockInserter();
+			await navigation.addLinkClose();
+
+			/**
+			 * TODO: This is not desired behavior. Ideally the
+			 * Appender should be focused again since it opened
+			 * the link control.
+			 * IMPORTANT: This check is not to enforce this behavior,
+			 * but to make sure focus is kept nearby until we are able
+			 * to send focus to the appender.
+			 */
+			await expect( navBlock ).toBeFocused();
+		} );
 
-		/**
-		 * Test: Creating a link sends focus to the newly created navigation link item
-		 */
-		await pageUtils.pressKeys( 'ArrowDown' );
+		// eslint-disable-next-line playwright/expect-expect
+		test( 'Focus management when creating navigation links', async ( {
+			page,
+			pageUtils,
+			navigation,
+		} ) => {
+			/**
+			 * Test: Creating a link sends focus to the newly created navigation link item
+			 */
+			await pageUtils.pressKeys( 'ArrowDown' );
+			await navigation.useBlockInserter();
+			await navigation.addPage( 'Cat' );
+			await navigation.checkLabelFocus( 'Cat' );
```

Comment:
```suggestion
			await navigation.addPage( 'Cat' );
```

The `addPage` method also [contains the label focus check](https://github.com/WordPress/gutenberg/blob/59a230c1e692ed78d0abfebce3c6ff5b429c1a4b/test/e2e/specs/editor/blocks/navigation.spec.js#L689), so there should be no need to run checkLabelFocus here.